plugins/modules/zos_find.py

```diff
@@ -336,7 +336,8 @@ def data_set_filter(module, pds_paths, patterns):
         rc, out, err = _dls_wrapper(pattern, list_details=True)
         if rc != 0:
             if "BGYSC1103E" in err:
-                return filtered_data_sets
+                # return filtered_data_sets
+                continue
```

Comment:
At first I agreed with this, but then it appears if when we do encounter a BGYSC1103E we will continue and go  onto hit the fail_json code.It sort of looks like we don't want to fail at this point and instead review the remaining results to see if we can find something.

Your test case seems to cover the scenario below but still feel there is something off.

In the users example here, returned 0 because we failed at the first one, when we could have returned 2 found.
```
pds_patterns:
- IMSTEST.DS.NOTEXIST
- IMSTEST.USER.A
- IMSTEST.USER.B
```
and here 1 was returned when it could have been 2 as well.
```
- IMSTEST.USER.A
- IMSTEST.DS.NOTEXIST
- IMSTEST.USER.B
```

Reply:
Ah, you are right, sorry about that @richp405, the loop will continue, I think i had done 1 too many code reviews that day.